app/src/main/java/edu/mit/mitmobile2/tour/activities/TourSelfGuidedActivity.java

```diff
@@ -115,4 +118,33 @@ public void showTourDetailActivity(String description) {
         intent.putExtra(Constants.Tours.TOUR_DETAILS_KEY, description);
         startActivity(intent);
     }
+
+    @Override
+    public void showMainLoopFragment(int currentStopNum) {
+        TourStopsFragment tourStopsFragment = new TourStopsFragment();
+        Bundle args = new Bundle();
+        args.putParcelable(Constants.TOURS, tour);
+        args.putInt(Constants.CURRENT_MAIN_LOOP_STOP, currentStopNum);
+        tourStopsFragment.setArguments(args);
+        getFragmentManager().beginTransaction().replace(R.id.tour_frame, tourStopsFragment).commit();
+    }
+
+    @Override
+    public void showSideTripFragment(MITTourStop mitTourStop) {
+        TourStopViewPagerFragment tourStopViewPagerFragment = TourStopViewPagerFragment.newInstance(mitTourStop);
+        Bundle args = new Bundle();
+        args.putParcelable(Constants.TOUR_STOP, mitTourStop);
+        tourStopViewPagerFragment.setArguments(args);
+        getFragmentManager().beginTransaction().replace(R.id.tour_frame, tourStopViewPagerFragment).commit();
+    }
+
+    @Override
+    public void setMainLoopActionBarTitle(int mainLoopStopNum, int mainLoopStopsSize) {
+        setTitle(getResources().getString(R.string.stop_nav_main_loop) + " " + mainLoopStopNum + " of " + mainLoopStopsSize);
```

Comment:
You can pass values into a string resource to avoid this. See the "Formatting strings" section of http://developer.android.com/guide/topics/resources/string-resource.html